src/main/java/Duke.java

```diff
@@ -1,10 +1,389 @@
+    private void markTaskAsDone(int index) {
+        Task task = getTask(index);
+        task.markAsDone();
+        printHorizontalLine();
+        printWithIndent(" Nice! I've marked this task as done:");
+        printWithIndent("   " + task);
+        printHorizontalLine();
+    }
+
+    // Marks the i-th task in the task list as not done, where i is the index.
+    private void markTaskAsNotDone(int index) {
+        Task task = getTask(index);
+        task.markAsNotDone();
+        printHorizontalLine();
+        printWithIndent(" OK, I've marked this task as not done yet:");
+        printWithIndent("   " + task);
+        printHorizontalLine();
+    }
+
+    // Prints the task list.
+    private void printTaskList() {
+        int numOfTasks = taskList.size();
+        printHorizontalLine();
+        printWithIndent(" Here are the tasks in your list:");
+
+        for (int index = 1; index <= numOfTasks; index++) {
+            printWithIndent(" " + index + "." + taskList.get(index - 1));
+        }
+
+        printHorizontalLine();
+    }
+
+    // Deletes the i-th task in the task list, where i is the index.
+    private void deleteTask(int index) {
+        Task deletedTask = taskList.remove(index - 1);
+        int numTasks = numTasks();
+        printHorizontalLine();
+        printWithIndent(" Noted. I've removed this task:");
+        printWithIndent("   " + deletedTask);
+        printWithIndent(" Now you have " + numTasks + " tasks in the list.");
+        printHorizontalLine();
+    }
+
+    // Handles the command input by the user.
+    private void handleCommand(String command) throws UnknownCommandException {
+        try {
+            if (isCommandMark(command)) {
+                markTaskAsDone(getMarkUnmarkDeleteIndex(command));
+            } else if (isCommandUnmark(command)) {
+                markTaskAsNotDone(getMarkUnmarkDeleteIndex(command));
+            } else if (isCommandList(command)) {
+                printTaskList();
+            } else if (isCommandTodo(command)) {
+                addTodo(getTodoDescription(command));
+            } else if (isCommandDeadline(command)) {
+                addDeadline(getDeadlineDescription(command), getDeadlineBy(command));
+            } else if (isCommandEvent(command)) {
+                addEvent(getEventDescription(command), getEventFrom(command), getEventTo(command));
+            } else if (isCommandDelete(command)) {
+                deleteTask(getMarkUnmarkDeleteIndex(command));
+            } else {
+                throw new UnknownCommandException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+        } catch (EmptyDescriptionException e) {
+            printHorizontalLine();
+            System.out.println("     " + e.getMessage());
+            printHorizontalLine();
+        }
+    }
+
+    // Takes inputs from user until bye has been input.
+    private void takeInputsUntilBye() {
```

Comment:
method name although is intuitive , but it can be made a bit more general e.g `userInput`